### PR TITLE
Add preset for site relation

### PR DESCRIPTION
**Pull request by kepta**
Adds a preset for site relation
Create
* site.json
* site/heritage.json
* site/historic.json
* site/parking.json

### CODE REVIEW

**Comment by kepta** [comment]
@jfirebaugh @bhousel review please?

**Comment by pnorman** [comment]
It's failing CI when running `make`

File: data/presets/presets/site/parking.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "geometry": [
+        "point",
+        "vertex",
+        "area"
+    ],
+    "tags": {
+        "site": "parking"
+    },
+    "terms": [],
+    "name": "Parking Site"
```
Comment (pnorman) [inline data/presets/presets/site/parking.json]: Is this going to be clear enough to users so they don't use it inappropriately?

**Comment by kepta** [comment]
@pnorman should I change it someway?

**Comment by bhousel** [comment]
What you made would be good for a normal preset, but Site Relations are special.  They're not a typical `point/area/vertex` kind of thing that users would add to the map.

Instead, try to copy what the Boundary Relation preset does:  [`boundary.json`](https://github.com/openstreetmap/iD/blob/master/data/presets/presets/type/boundary.json).

The icon for this new preset can be just be the generic `relation` icon.

**Comment by kepta** [comment]
Well I am still confused about the preset part of iD, I did go through the basic readme in the preset part. 
Is there a doc which details the implementation ?

**Comment by bhousel** [comment]
The implementation is indeed confusing.  Thankfully contributors don't need to know anything about the implementation to add presets to iD.  But if you want to dive in, here is a cheetsheet:

This class encapsulates the whole preset subsystem.
* [`js/id/presets.js`](https://github.com/openstreetmap/iD/blob/master/js/id/presets.js) 

These files implement the different preset objects.
* [`js/id/presets/category.js`](https://github.com/openstreetmap/iD/blob/master/js/id/presets/category.js) 
* [`js/id/presets/collection.js`](https://github.com/openstreetmap/iD/blob/master/js/id/presets/collection.js) 
* [`js/id/presets/field.js`](https://github.com/openstreetmap/iD/blob/master/js/id/presets/field.js) 
* [`js/id/presets/preset.js`](https://github.com/openstreetmap/iD/blob/master/js/id/presets/preset.js)  

UI for presets is separate from all of the above and implemented here.  Everything UI will be really confusing until you understand the [D3.js](http://d3js.org/) way of doing things.
* [`js/id/ui/entity_editor.js`](https://github.com/openstreetmap/iD/blob/master/js/id/ui/entity_editor.js) - everything in sidebar
* [`js/id/ui/preset.js`](https://github.com/openstreetmap/iD/blob/master/js/id/ui/preset.js) - top part of sidebar including icon and preset fields
* [`js/id/ui/preset_icon.js`](https://github.com/openstreetmap/iD/blob/master/js/id/ui/preset_icon.js) - just the icon
* [`js/id/ui/preset_list.js`](https://github.com/openstreetmap/iD/blob/master/js/id/ui/preset_list.js) - the list of presets and categories

The different preset field types (`access`,`check`,`combo`,`input`,etc.) UI are all located under this folder:
* [`js/id/ui/preset/`](https://github.com/openstreetmap/iD/tree/master/js/id/ui/preset)

Hope this is helpful!

**Comment by bhousel** [comment]
Looks good thank you!, just merged as 6a49443

I also added this preset to `defaults.json` so it will show up in this menu:

screenshot 2016-02-01 14 13 49

**Pull request closed without merging**